modules/ml/src/main/java/org/apache/ignite/ml/trees/trainers/columnbased/contsplitcalcs/VarianceSplitCalculator.java

```diff
@@ -0,0 +1,176 @@
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.ml.trees.trainers.columnbased.contsplitcalcs;
+
+import java.io.IOException;
+import java.io.ObjectInput;
+import java.io.ObjectOutput;
+import java.util.PrimitiveIterator;
+import java.util.stream.DoubleStream;
+import org.apache.ignite.ml.trees.ContinuousRegionInfo;
+import org.apache.ignite.ml.trees.ContinuousSplitCalculator;
+import org.apache.ignite.ml.trees.trainers.columnbased.vectors.ContinuousSplitInfo;
+import org.apache.ignite.ml.trees.trainers.columnbased.vectors.SplitInfo;
+
+/**
+ * Calculator of variance in a given region.
+ */
+public class VarianceSplitCalculator implements ContinuousSplitCalculator<VarianceSplitCalculator.VarianceData> {
+    /**
+     * Data used in variance calculations.
+     */
+    public static class VarianceData extends ContinuousRegionInfo {
+        /** Mean value in a given region. */
+        double mean;
+
+        /**
+         * @param var Variance in this region.
+         * @param size Size of data for which variance is calculated.
+         * @param mean Mean value in this region.
+         */
+        public VarianceData(double var, int size, double mean) {
+            super(var, size);
+            this.mean = mean;
+        }
+
+        /**
+         * No-op constructor. For serialization/deserialization.
+         */
+        public VarianceData() {
+            // No-op.
+        }
+
+        /** {@inheritDoc} */
+        @Override public void writeExternal(ObjectOutput out) throws IOException {
+            super.writeExternal(out);
+            out.writeDouble(mean);
+        }
+
+        /** {@inheritDoc} */
+        @Override public void readExternal(ObjectInput in) throws IOException, ClassNotFoundException {
+            super.readExternal(in);
+            mean = in.readDouble();
+        }
+
+        /**
+         * Returns mean.
+         */
+        public double mean() {
+            return mean;
+        }
+    }
+
+    /** {@inheritDoc} */
+    @Override public VarianceData calculateRegionInfo(DoubleStream s, int size) {
+        PrimitiveIterator.OfDouble itr = s.iterator();
+        int i = 0;
+
+        double mean = 0.0;
+        double m2 = 0.0;
+
+        while (itr.hasNext()) {
```

Comment:
Please add some comment about this while.

Reply:
Done.